src/Language/PureScript/Constants/Prelude.hs

```diff
@@ -273,6 +273,12 @@ pattern DataSymbol = ModuleName "Data.Symbol"
 pattern IsSymbol :: Qualified (ProperName 'ClassName)
 pattern IsSymbol = Qualified (Just DataSymbol) (ProperName "IsSymbol")
 
+pattern DataReflectable :: ModuleName
+pattern DataReflectable = ModuleName "Data.Reflectable"
+
+pattern IsReflectable :: Qualified (ProperName 'ClassName)
+pattern IsReflectable = Qualified (Just DataReflectable) (ProperName "IsReflectable")
```

Comment:
I was thinking, to reduce breakage, we would initially make `IsSymbol` extend `IsReflectable`. 
```purescript
class IsReflectable sym String => IsSymbol sym

reflectSymbol :: forall sym. IsSymbol sym => Proxy sym -> String
reflectSymbol = reflectType
```

As for whether to deprecate and then later remove `IsSymbol` entirely, I'm not sure.

Reply:
Technically still a breaking change, as users can currently implement `IsSymbol` instances themselves:

```purs
foreign import data MySymbol :: Symbol

instance IsSymbol MySymbol where
  reflectSymbol _ = "a deeply bad idea?"
```

I don't know if anyone's done this, but we've left the footgun lying around, so to speak.

Reply:
Ugh...

Reply:
> but we've left the footgun lying around, so to speak.

I do feel like the responsible choice to do here is to go ahead and remove the footgun entirely since it's not really the intended behaviour. I'm keen on deprecating `IsSymbol` if this PR lands and we end up removing it down the line entirely, perhaps along with replacing `Symbol` with `String`.

Reply:
I think the larger issue is that kinds are currently open. We'd have to make these specific kinds closed to completely remove this footgun. Because once this PR lands, one could also do something like `foreign import data Foo :: Integer`, which creates a new variant of the same footgun.

Reply:
It sounds like we should do the following:
- now 
    - support both `IsSymbol` and `IsReflectable` separately (so no breaking changes are made)
    - deprecate `IsSymbol`?
- later in a breaking release
    - remove `IsSymbol`
    - rename the kind `Symbol` to `String`
- later (at some point in the future)
    - remove the footgun